plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteInsertBatchImpl.java

```diff
@@ -157,56 +161,38 @@ StringBuilder prepareQueryForStatement(
                 attributeName = DBUtils.getObjectFullName(table.getDataSource(), attribute, DBPEvaluationContext.DML);
             }
             query.append(attributeName);
+            usedAttributes.add(i);
         }
         query.append(")\n\tVALUES "); //$NON-NLS-1$
-        StringBuilder valuesPart = new StringBuilder(64);
-        valuesPart.append("(");
-        hasKey = false;
-        boolean skipBindValues = CommonUtils.toBoolean(options.get(DBSDataManipulator.OPTION_SKIP_BIND_VALUES));
-        for (int i = 0; i < attributes.length; i++) {
-            DBSAttributeBase attribute = attributes[i];
-            if (DBUtils.isPseudoAttribute(attribute) || (!useMultiRowInsert && (!allNulls && DBUtils.isNullValue(attributeValues[i])))) {
-                continue;
-            }
-            if (hasKey) valuesPart.append(","); //$NON-NLS-1$
-            hasKey = true;
+        
+        int usedAttributesArr[] = usedAttributes.stream().mapToInt(Integer::intValue).toArray();
 
-            DBDValueHandler valueHandler = handlers[i];
-            
-            if (valueHandler instanceof DBDValueBinder) {
-                valuesPart.append(((DBDValueBinder) valueHandler).makeQueryBind(attribute, attributeValues[i]));
-            } else {
-                valuesPart.append("?"); //$NON-NLS-1$
-            }
-        }
-        valuesPart.append(")"); //$NON-NLS-1$
-        if (useMultiRowInsert) {
-            for (int i = 0; i < attributeValues.length / attributes.length; i++) {
-                if (i != 0) {
-                    query.append(",");
+        StringJoiner valuesPart = new StringJoiner(","); //$NON-NLS-1$
+        boolean skipBindValues = CommonUtils.toBoolean(options.get(DBSDataManipulator.OPTION_SKIP_BIND_VALUES));
+        for (int i = 0; i < attributeValues.length / attributes.length; i++) {
+            StringJoiner rowValuesPart = new StringJoiner(",", "(", ")"); //$NON-NLS-1$
```

Comment:
The `$NON-NLS-1$` comment only covers the first string literal `","`. Also, it's obsolete and you can remove it entirely.

Reply:
Done